ibis/backends/druid/__init__.py

```diff
@@ -130,11 +133,18 @@ def get_schema(
         catalog: str | None = None,
         database: str | None = None,
     ) -> sch.Schema:
-        return self._get_schema_using_query(
+        query = (
             sg.select(STAR)
             .from_(sg.table(table_name, db=database, catalog=catalog))
             .sql(self.dialect)
         )
+        try:
+            schema = self._get_schema_using_query(query)
+        except PyDruidProgrammingError as e:
+            if re.search(r"\bINVALID_INPUT\b", str(e)):
+                raise com.TableNotFound(table_name) from e
```

Comment:
```suggestion
                raise com.TableNotFound(table_name) from e
            raise
```

Remember that you have to continue to `raise`. There are other reasons besides the missing table that these errors may occur.